src/app_service/service/message/service.nim

```diff
@@ -188,9 +193,6 @@ QtObject:
     let pinnedMsgCursorValue = if (pinnedMsgCursor.isFetchable()): pinnedMsgCursor.getValue() else: CURSOR_VALUE_IGNORE
 
     if(msgCursorValue == CURSOR_VALUE_IGNORE and pinnedMsgCursorValue == CURSOR_VALUE_IGNORE):
-      # it's important to emit signal in case we are not fetching messages, so we can update the view appropriatelly.
-      let data = MessagesLoadedArgs(chatId: chatId)
-      self.events.emit(SIGNAL_MESSAGES_LOADED, data)
```

Comment:
Why did you remove this?

Reply:
It looks like some legacy part of the code which does not need now.
When we open a new chat, we receive a few `SIGNAL_MESSAGES_LOADED` events, this one the first and the second event - with the initial messages.
SIGNAL_MESSAGES_LOADED event subscriber mark `firstUnseenMessageLoaded` as true, but in reality, we didn't get the messages. So we started displaying chat UI while messages are still loading and we can observe behavior, when we have chat name/icon, and fetch content, but after a few seconds, messages are loading.

I see, that this event emitted again from this place only when we scrolled to the top of available messages.
I did not find what it breaks

Reply:
I **think** it used to make it hide the little loader at the top of a chat when we start fetching messages. Do you see that it stays forever now, especially when you scroll to the top of a channel?

Reply:
based on what I see, it is just stopping processing new messages fetch when we do not need it.
It sends empty data and doesn't impact the UI (at least I did not find).
SIGNAL_MESSAGES_LOADED will be sent second time, but with messages (or without, if there are no messages)